scripts/azureml-assets/azureml/assets/model/model_utils.py

```diff
@@ -151,3 +152,37 @@ def prepare_model(spec_path, model_config, registry_name, temp_dir):
     except Exception as e:
         logger.log_error(f"prepare model failed for {spec_path}. Error {e}")
         return None, False
+
+
+def update_model_metadata(
+        model_name: str,
+        model_version: str,
+        registry_name: str,
+        update: AssetVersionUpdate):
+    """Update the mutable metadata of already registered Model."""
+    try:
+        ml_client = MLClient(credential=AzureCliCredential(), registry_name=registry_name)
```

Comment:
Consider providing an optional ml_client arg to the function and only create the MLClient if the arg is none, or just create the MLClient outside of this function entirely. Creating MLClient is somewhat expensive and results in multiple calls to retrieve the token and call registry discovery.